src/api/routes/v1/get-network.ts

```diff
@@ -0,0 +1,237 @@
+ * @returns The crawl data from the node.
+ */
+async function fetchCrawls(host: string): Promise<Crawl> {
+  const promises: Array<Promise<Crawl | undefined>> = []
+  for (const port of CRAWL_PORTS) {
+    promises.push(crawlNode(host, port))
+  }
+  return any(promises)
+}
+
+/**
+ * Checks whether the UNL of the node is recorded in the networks table. If so,
+ * then it returns the network associated with the UNL.
+ *
+ * @param unl - The UNL of the node.
+ * @returns Whether the UNL of the node has been seen before.
+ */
+async function getNetworkFromUNL(unl: string): Promise<string | undefined> {
+  const result = await query('networks')
+    .select('id')
+    .where('unls', 'like', `%${unl}%`)
+  return result.length > 0 ? result[0].id : undefined
+}
+
+/**
+ * Checks whether the public key of the node is recorded in the crawls table. If
+ * so, then it returns the network associated with the public key.
+ *
+ * @param publicKey - The public key of the node.
+ * @returns The network associated with the public key of the node. Undefined
+ * if the public key has not been seen by the network.
+ */
+async function getNetworkFromPublicKey(
+  publicKey: string,
+): Promise<string | undefined> {
+  const result = await query('crawls')
+    .select('public_key', 'networks')
+    .where('public_key', '=', publicKey)
+  return result.length > 0 ? result[0].networks : undefined
+}
+
+/**
+ * Adds the node (and its corresponding network) to the networks table.
+ *
+ * @param url - The URL endpoint of the node.
+ * @param unl - The UNL of the node.
+ * @returns The ID of the new network.
+ */
+async function addNode(url: string, unl: string | null): Promise<string> {
+  const newNetwork = (Number(maxNetwork) + 1).toString()
+  maxNetwork += 1
+
+  const network: Network = {
+    id: newNetwork,
+    entry: url,
+    port: 51235,
+    unls: unl ? [unl] : [],
+  }
+  await query('networks').insert({
+    ...network,
+    unls: network.unls.join(','),
+  })
+
+  const crawler = new Crawler()
+  void crawler.crawl(network)
+
+  return newNetwork
+}
+
+/**
+ * Fetch network ID from network entry. If the network doesn't exist, adds it with a new ID.
+ *
+ * @param req - Express request.
+ * @param res - Express response.
+ * @returns The Express response.
+ */
+export default async function getNetwork(
+  req: Request,
+  res: Response,
+): Promise<Response> {
+  try {
+    const { entryUrl } = req.params
+
+    // fetch crawl
+    const crawl = await fetchCrawls(entryUrl)
+
+    // check UNL
+    const { node_unl } = crawl
+    if (node_unl != null) {
+      const unlNetwork = await getNetworkFromUNL(node_unl)
+      // eslint-disable-next-line max-depth -- Necessary here
+      if (unlNetwork != null) {
+        return res.send({
+          result: 'success',
+          network: unlNetwork,
+        })
+      }
+    }
+
+    // check if node public key is already recorded
+    const { public_key } = crawl.this_node
```

Comment:
is `this_node` guaranteed to exist? If not this will error:
```js
const foo = {}
const { does_not_exist } = foo.not_a_field
```
Because that the same as
```js
const does_not_exists = foo.not_a_field.does_not_exist
``` 

and JS doesn't let you call `.does_not_exist` on undefined.

Reply:
If it's a valid crawl, it should. If not, I believe it'll error in the [`crawlNode` function in the crawler](https://github.com/ripple/validator-history-service/blob/95ba58da44d82308e34c169e79cdad109528ab50/src/crawler/network.ts#L41-L95), where the crawl is fetched.